src/metabase/api/pulse.clj

```diff
@@ -31,7 +31,11 @@
    channels [Required ArrayOfMaps]}
   ;; prevent more than 5 cards
   ;; limit channel types to :email and :slack
-  (->500 (pulse/create-pulse name *current-user-id* (filter identity (map :id cards)) channels)))
+  (->500 (pulse/create-pulse name *current-user-id*
+                             (for [{:keys [id]}  cards
+                                   :when         id]
+                               id)
```

Comment:
i'm fine with the way you rewrote this using `for`, so this is just a comment for fun to talk about syntax, but i actually think the original code is a more natural way to express the value in this particular case.

i think that using `for` is a better pattern in many cases, so it makes sense that it should be the default approach, but i think using it here is harder to grok.